### DISCUSSION TITLE
Median risk for Cox regression

### DISCUSSION THREAD

**Discussion by MuradOmarov** [CamDavidsonPilon/lifelines#1423] [Q&A]
Hello! Can you please tell me is there a way to plot median risk from train data population with predicted survival function?

**Comment by CamDavidsonPilon** [comment]
Hi, I'm not sure what you mean? Can you explain in more detail / a different way?

**Reply by MuradOmarov** [reply]
I'm sorry for the inaccurate wording. 
I trained Cox regression with large dataset (one population) and now I want to plot individual survival functions for test data (from the same population). But along with these curves I want to draw median risk/survival course (preferably with confidence intervals) from whole population. And my question is what is the best way to do this? Is it relevant for this type of model?
Should I first calculate the survival functions for the entire dataset (train + test) and then calculate the median manually?
Or is there a better way?
Thanks in advance!

**Comment by CamDavidsonPilon** [comment]
I would probably train two Cox models, if I had to do this (though I think it's a bit strange tbh). First model, `cox1`, would be fitted to only the training data. Second model, `cox2`, to the entire data (train + test). 

1. Use `cox1.predict_survival_functions(test_data)` to get survival curves and plot them.
2. Median survival curve is ambiguous: do you mean median covariates, or median survival probabilities. If the former, then you can use `cox2.predict_survival_function(cox2._central_values)`. If the latter, use `cox2.predict_survival_function(all_data).median(axis=0)`.

**Reply by MuradOmarov** [reply]
Thank you very much, Cameron!
